mautic/mautic
# Catch exception if preview fails

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | N
| Automated tests included? | N
| Related user documentation PR URL | /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | /
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

We discovered in our logs this error:
```
exception:  "WideImage_InvalidImageSourceException: File '/var/www/hosted/wilson/public_html/images/812cd24e605deca2197f8c4ce77c4c37.gif' appears to be an invalid image source. in /usr/local/share/mautic/production/releases/2018.12.01-1/app/bundles/CoreBundle/Assets/js/libraries/ckeditor/filemanager/connectors/php/inc/wideimage/lib/WideImage.php:207
Stack trace:
#0 /usr/local/share/mautic/production/releases/2018.12.01-1/app/bundles/CoreBundle/Assets/js/libraries/ckeditor/filemanager/connectors/php/inc/wideimage/lib/WideImage.php(168): WideImage::loadFromFile('/var/www/hosted...')
#1 /usr/local/share/mautic/production/releases/2018.12.01-1/app/bundles/CoreBundle/Assets/js/libraries/ckeditor/filemanager/connectors/php/filemanager.class.php(1451): WideImage::load('/var/www/hosted...')
#2 /usr/local/share/mautic/production/releases/2018.12.01-1/app/bundles/CoreBundle/Assets/js/libraries/ckeditor/filemanager/connectors/php/filemanager.class.php(965): Filemanager->get_thumbnail('/var/www/hosted...')
#3 /usr/local/share/mautic/production/releases/2018.12.01-1/app/bundles/CoreBundle/Assets/js/libraries/ckeditor/filemanager/connectors/php/filemanager.php(103): Filemanager->preview(true)
#4 {main}" 
```
The image `812cd24e605deca2197f8c4ce77c4c37.gif` is here for testing:

![broken_img-omg](https://user-images.githubusercontent.com/1235442/50006726-fa16f100-ffae-11e8-95d8-bbfeaab9584e.gif)

It can be uploaded but then when displayed in the file manager it throws the error. It looks like this in the file manager:

![screenshot 2018-12-13 at 16 11 26](https://user-images.githubusercontent.com/1235442/50006771-1dda3700-ffaf-11e8-910e-56b6c387e4a0.png)

The fix only catches the exception and set it to the error handler.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Go to the Landing Page Builder and upload the `broken_img-omg` image linked above.
2. Go to the file manager.
3. Open the browser dev tools.
4. Refresh the file manager and notice in the Network tab that the image throws 500 response code.
5. If you open the failed request in a new tab on URL like: `[your mautic domain]/app/bundles/CoreBundle/Assets/js/libraries/ckeditor/filemanager/connectors/php/filemanager.php?mode=preview&path=%2Fbroken_img-omg.gif&1544793990&thumbnail=true` then you'll see the classic Mautic 500 error: 

#### Steps to test this PR:
1. Refresh the tab with the 500 error and you'll see 200 code instead and this message: `{"Error":"File '\/Users\/jan\/dev\/mautic-hosted\/media\/images\/broken_img-omg.gif' appears to be an invalid image source.","Code":"-1","Properties":{"Date Created":null,"Date Modified":null,"Height":null,"Width":null,"Size":null}}`
